Gems/Atom/Feature/Common/Code/Source/CoreLights/ProjectedShadowmapsPass.cpp

```diff
@@ -210,6 +270,44 @@ namespace AZ
             return ShadowmapPass::CreateWithPassRequest(passName, passData);
         }
 
+        void ProjectedShadowmapsPass::CreateClearShadowDrawPacket()
+        {
+            // Force load of shader to clear shadow maps.
+            const AZStd::string clearShadowShaderFilePath = "Shaders/Shadow/ClearShadow.azshader";
```

Comment:
I think I noticed that the clear shadow shader was doing a fullscreen triangle VS approach. It might be worth doing the clear with a render target clear since the hardware can do faster clears operating over the metadata (as opposed to per-pixel). Maybe there's a reason this isn't possible that I'm missing?

Reply:
Doing the shader approach because certain platforms *cough* metal *cough* can't do clears of regions of render targets. Since the shadows use an atlas, it's possible some items in the atlas are cached and some aren't, so only certain viewports need to be cleared. While there is support for this kind of thing in DirectX and Vulkan, we haven't plumbed it into our RHI, and it's not support in metal at all.

Reply:
I do regular full clears in cases where an atlas has no cached shadows, so the extra cost is only when there's a mix.